src/Program.cs

```diff
@@ -26,21 +28,51 @@ static void Main(string[] args)
                 if (File.Exists(inputPath))
                 {
                     CommandLineUtils.CreateOutputDirectory(outputPath);
-                    HtmlProcessor.ProcessFile(inputPath, outputPath);
+
+                    // Check file extension to decide whether to use MdProcessor or HtmlProcessor
+                    string fileExtension = Path.GetExtension(inputPath);
+                    if (fileExtension == ".md")
+                    {
+                        MdProcessor.ProcessFile(inputPath, outputPath);
+                    }
+                    else if (fileExtension == ".txt")
+                    {
+                        HtmlProcessor.ProcessFile(inputPath, outputPath);
+                    }
+                    else
+                    {
+                        CommandLineUtils.Logger($"Unsupported file format for input path: {inputPath}");
```

Comment:
This is check is unnecessary, because it already exists in `HtmlProcessor`. But you should modify it to check for `.md` as well.